6 add unit test for crud user and room

Brawdunoir opened the pull request.

Brawdunoir committed:
Declare router outside of main package
Get VERSION from env

Brawdunoir committed:
Declare router outside of main package
Get VERSION from env

Brawdunoir committed:
Add intermediary image for development
Remove port to host for the db

Brawdunoir committed:
Add intermediary image for development
Remove port to host for the db

Brawdunoir committed:
Add a docker-compose file for testing purposes

Brawdunoir committed:
Add a docker-compose file for testing purposes

Brawdunoir committed:
Change to tty to keep open api for tests

Brawdunoir committed:
Change to tty to keep open api for tests

Brawdunoir committed:
Add empty router to populate with routes in the SetupRouter function

Brawdunoir committed:
Add empty router to populate with routes in the SetupRouter function

Brawdunoir committed:
Add Get and Create tests for user

Brawdunoir committed:
Add Get and Create tests for user

Brawdunoir committed:
Use random suffix for temp db volume mount

Brawdunoir committed:
Use random suffix for temp db volume mount

Brawdunoir committed:
Harmonize error

Brawdunoir committed:
Harmonize error

Brawdunoir committed:
Allow response generalization for object creation

Brawdunoir committed:
Allow response generalization for object creation

Brawdunoir committed:
Realize that Read, Update and Delete is the same code for tests

Brawdunoir committed:
Realize that Read, Update and Delete is the same code for tests

Brawdunoir committed:
Factorize variables

Brawdunoir committed:
Factorize variables

Brawdunoir committed:
Add update tests for user and patch api

Brawdunoir committed:
Add update tests for user and patch api

Brawdunoir committed:
Add delete tests for user and patch api

Brawdunoir committed:
Add delete tests for user and patch api

Brawdunoir committed:
Remove lint typecheck

Brawdunoir committed:
Remove lint typecheck

Brawdunoir committed:
Add room tests

Brawdunoir committed:
Add room tests

Brawdunoir committed:
Lint

Brawdunoir committed:
Lint

Brawdunoir committed:
Lint

Brawdunoir committed:
Lint

Brawdunoir committed:
Return code 500 if the version is not set

Brawdunoir committed:
Return code 500 if the version is not set

vazanoir commented on tests/room_test.go:
```diff
@@ -0,0 +1,86 @@
+	test := utils.TestCreate{
+		SubTests: []utils.SubTest{
+			{Name: "Success", Request: utils.Request{Method: method, URL: roomURL, Body: `{"name":"test"}`}, ResponseCode: http.StatusCreated, ResponseBodyRegex: `{"id":"\d+"}`},
+			{Name: "Empty body", Request: utils.Request{Method: method, URL: roomURL, Body: ``}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":".+"}`},
```
```suggestion
			{Name: "Empty body", Request: utils.Request{Method: method, URL: roomURL, Body: ``}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":".+"}`},
			{Name: "Empty json", Request: utils.Request{Method: method, URL: roomURL, Body: `{}`}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":".+"}`},
```

vazanoir replied:
Add it for every test like this

vazanoir commented on tests/room_test.go:
```diff
@@ -0,0 +1,86 @@
+			{Name: "Success", Request: utils.Request{Method: method, URL: roomURL, Body: `{"name":"test"}`}, ResponseCode: http.StatusCreated, ResponseBodyRegex: `{"id":"\d+"}`},
+			{Name: "Empty body", Request: utils.Request{Method: method, URL: roomURL, Body: ``}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":".+"}`},
+			{Name: "Bad name key", Request: utils.Request{Method: method, URL: roomURL, Body: `{"wrongkey":"test"}`}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":".+"}`},
+			{Name: "Bad name value", Request: utils.Request{Method: method, URL: roomURL, Body: `{"name":""}`}, ResponseCode: http.StatusBadRequest, ResponseBodyRegex: `{"error":".+"}`},
```
Add tests for int/object/(nil/null) in values

vazanoir reviewed (changes requested):
Moke rooms and users for every crud tests

Brawdunoir committed:
Allow all origins cors

Brawdunoir committed:
Merge pull request #35 from Brawdunoir/32-fix-access-control-allow-origin-1

Allow all origins cors

Brawdunoir committed:
Fix CORS on client side. Attempt #3 (#36)

Brawdunoir committed:
Add empty json test

Co-authored-by: Vazanoir <vazanoir@outlook.fr>

Brawdunoir committed:
Add empty json test

Co-authored-by: Vazanoir <vazanoir@outlook.fr>

Brawdunoir committed:
Add more tests depending of the body type
Add get test after update and delete.
We also assume that that get is okay (since we test it).
We assume that create is okay (since we can get).

Brawdunoir committed:
Add more tests depending of the body type
Add get test after update and delete.
We also assume that that get is okay (since we test it).
We assume that create is okay (since we can get).

Brawdunoir committed:
Merge branch '6-add-unit-test-for-crud-user-and-room' of github.com:Brawdunoir/dionysos-server into 6-add-unit-test-for-crud-user-and-room

Brawdunoir committed:
Correct wrong copy/paste

Brawdunoir committed:
Correct wrong rebase

WhiteDiamondz commented on tests/room_test.go:
```diff
@@ -0,0 +1,96 @@
+	test.Run(t)
+}
+
+// TestGetRoom test the GetRoom function.
```
```suggestion
// TestGetRoom tests the GetRoom function.
```

WhiteDiamondz commented on tests/room_test.go:
```diff
@@ -0,0 +1,96 @@
+	test.Run(t)
+}
+
+// TestUpdateRoom test the UpdateRoom function.
```
```suggestion
// TestUpdateRoom tests the UpdateRoom function.
```

WhiteDiamondz commented on tests/room_test.go:
```diff
@@ -0,0 +1,96 @@
+	test.Run(t)
+}
+
+// TestDeleteRoom test the DeleteRoom function.
```
```suggestion
// TestDeleteRoom tests the DeleteRoom function.
```

WhiteDiamondz commented on tests/user_test.go:
```diff
@@ -0,0 +1,96 @@
+	test.Run(t)
+}
+
+// TestGetUser test the GetUser function.
```
```suggestion
// TestGetUser tests the GetUser function.
```

WhiteDiamondz commented on tests/user_test.go:
```diff
@@ -0,0 +1,96 @@
+	test.Run(t)
+}
+
+// TestUpdateUser test the UpdateUser function.
```
```suggestion
// TestUpdateUser tests the UpdateUser function.
```

WhiteDiamondz commented on tests/user_test.go:
```diff
@@ -0,0 +1,96 @@
+	test.Run(t)
+}
+
+// TestDeleteUser test the DeleteUser function.
```
```suggestion
// TestDeleteUser tests the DeleteUser function.
```

WhiteDiamondz commented on utils/tests/tests.go:
```diff
@@ -0,0 +1,59 @@
+	KeyCreated([]byte) (string, error)
+}
+
+// SubTest is an atomic test that include a request and its intended response.
```
```suggestion
// SubTest is an atomic test that includes a request and its intended response.
```

WhiteDiamondz commented on utils/tests/tests.go:
```diff
@@ -0,0 +1,59 @@
+	ResponseBodyRegex interface{}
+}
+
+// Request simple request to be send to the router.
```
```suggestion
// Request a simple request to be sent to the router.
```

WhiteDiamondz commented on utils/tests/tests_create.go:
```diff
@@ -0,0 +1,30 @@
+	SubTests []SubTest
+}
+
+// Runs a series of tests for a Create type endpoint.
```
```suggestion
// Runs a series of tests for the Create type endpoint.
```

WhiteDiamondz commented on utils/tests/tests_rud.go:
```diff
@@ -0,0 +1,45 @@
+	SubTests       []SubTest
+}
+
+// Runs a series of tests for a Get type endpoint.
```
```suggestion
// Runs a series of tests for the Get type endpoint.
```

WhiteDiamondz reviewed (changes requested):
Small fixes in comments, otherwise good to merge once corrected !

Brawdunoir committed:
Update tests/room_test.go

Co-authored-by: Fabien D <49648494+WhiteDiamondz@users.noreply.github.com>

Brawdunoir committed:
Update tests/room_test.go

Co-authored-by: Fabien D <49648494+WhiteDiamondz@users.noreply.github.com>

Brawdunoir committed:
FIx english in comments

Co-authored-by: Fabien D <49648494+WhiteDiamondz@users.noreply.github.com>

WhiteDiamondz reviewed (approved):
Good to go !
